src/main/scala/com/sandinh/paho/akka/MqttPubSub.scala

```diff
@@ -63,6 +63,14 @@ class MqttPubSub(cfg: PSConfig) extends FSM[PSState, Unit] {
   //reconnect attempt count, reset when connect success
   private[this] var connectCount = 0
 
+  /** Match one single level wild card character(i.e. + ).
+    * TODO: Wildcard matching needs to be extended to support multiple single level wildcards and combination of single level and multi level wildcards as specified in the specification: http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/csprd02/mqtt-v3.1.1-csprd02.html#_Toc385349843
+    * */
+  def matchSingleLevelWildcard(subTopic: String, actualTopic: String): Boolean = subTopic.replaceFirst("\\+",(actualTopic diff subTopic)) == actualTopic
```

Comment:
is it a good idea to use `diff` here, given that it removes characters without regard for the order in which they appear in either string ?